src/main/java/core/basesyntax/service/impl/FruitServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitService;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class FruitServiceImpl implements FruitService {
+
+    public FruitServiceImpl(FruitStorage fruitStorage) {
+    }
+
+    @Override
+    public void processTransactions(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            if (OperationStrategy.valueOf(transaction.parseOperation().toString()) == null) {
+                throw new NullPointerException("operation not exist");
+            }
+            OperationStrategy operation = OperationStrategy.valueOf(transaction.parseOperation()
+                        .toString().toUpperCase());
```

Comment:
Don't call operationStrategy as operation, when you have operation as enum - it's super confusing. NPE is never thrown manually - it usually just happens _to you_, and you take measures to mitigate this. ParseOperation already throws exception, if it can't turn String to enum. Do this instead:
Operation operation = Operation.parseOperation(transaction.getOperation()));
OperationStrategy strategy = OperationStrategy.valueOf(operation.name());